gapic/src/main/com/google/gapid/perfetto/models/GpuInfo.java

```diff
@@ -15,59 +15,63 @@
  */
 package com.google.gapid.perfetto.models;
 
-import static com.google.common.collect.Iterators.transform;
-import static com.google.common.collect.Iterators.unmodifiableIterator;
 import static com.google.gapid.util.MoreFutures.transform;
 
+import com.google.common.collect.Iterables;
 import com.google.common.util.concurrent.ListenableFuture;
 import com.google.gapid.models.Perfetto;
-import com.google.gapid.perfetto.models.GpuInfo.Gpu;
 
 import java.util.Collections;
-import java.util.Iterator;
-import java.util.SortedMap;
+import java.util.List;
 
 /**
  * Data about a GPU in the trace.
  */
-public class GpuInfo implements Iterable<Gpu> {
-  public static final GpuInfo NONE = new GpuInfo(Collections.emptySortedMap());
+public class GpuInfo {
+  public static final GpuInfo NONE = new GpuInfo(Collections.emptyList());
 
   private static final String MAX_DEPTH_QUERY =
-      "select ref, max(depth) + 1 from slices where ref_type = 'gpu' group by ref";
+      "select t.id, max(depth) + 1 " +
+      "from gpu_track t, gpu_slice s " +
+      "where t.id = s.track_id and t.scope = 'gpu_render_stage' " +
```

Comment:
Can we make this "where t.id = s.track_id " + "group by t.id " + "order by t.scope, t.id"?
We aren't doing anything special with the gpu_render_stage track.  If we don't limit the scope, we will be able to see other gpu tracks instead of having to change the UI every time we have a new type of track.

If we end up needing something special for gpu_render_stage, we can do refactor this so that we have a basic UI for all gpu tracks and a special case just for gpu_render_stage.

Reply:
No, i specifically don't want that. There already is a bug open to group slices vs counters in the UI. Since we have to update GAPID to use a newer version of the trace processor anyways for these future tracks to even be in the data, making the UI changes at the same time is no big deal.

Reply:
I agree with Pascal. That's why I didn't point out. The P0.9 is to get render stages show up.

Reply:
Maybe not in this PR, but if that's the argument, the code needs to be architect such that there is a list of scope it will display.  Adding a new GPU type should not require more than one or two lines of change.

Reply:
once we have more than one, i'm happy to make that change. Until we have more than one, it's not worth it.

Reply:
As I mentioned, we don't have to do it in this PR, but I want to note that we already have more than one.
On an R device, surface flinger register itself with data source name of "android.surfaceflinger.frame".  The trace processor only produces instant events at the moment.  So it still won't show with the change, but it would make it easier to test and develop.